network/p2p/connection/connector.go

```diff
@@ -95,8 +100,9 @@ func (l *Libp2pConnector) connectToPeers(ctx context.Context, peerIDs peer.IDSli
 	// create a channel of peer.AddrInfo as expected by the connector
 	peerCh := make(chan peer.AddrInfo, len(peerIDs))
 
-	// stuff all the peer.AddrInfo it into the channel
-	for _, peerID := range peerIDs {
+	// first shuffle, and then stuff all the peer.AddrInfo it into the channel.
+	// shuffling is not in place.
+	for _, peerID := range l.shuffler.Shuffle(peerIDs) {
```

Comment:
+1 on this. Unless the shuffling really needs to be deterministic, we should opt for the simpler/less code approach.

Reply:
https://github.com/onflow/flow-go/pull/4254/commits/5d7a9ab965d0b8496ec13f51286a0c638222427c

Reply:
Thank you for implementing the changes. The only remaining thing is to use the `Shuffle` function linked above instead of the `Shuffle` from `math/rand`. 
`math/rand` has some vulnerabilities and we don't want to rely on it in production code (also pointed out by the security audit). The tool I linked uses `crypto/rand` underneath.

Reply:
@yhassanzadeh13 I think @tarakby's last comment here is the last request in this PR.

Reply:
https://github.com/onflow/flow-go/pull/4254/commits/ebc3aebfff18c8d10ea217fef6b8a87763f7b430

Reply:
Thanks @yhassanzadeh13! I approve based on Peter's comment above that the PR is ready 👌🏼